src/app/payment/page.tsx

```diff
@@ -0,0 +1,29 @@
+import { getPayment } from '@/api/orderAPI';
+import { getAddresses } from '@/api/shippingAPI';
+import { dehydrate, HydrationBoundary, QueryClient } from '@tanstack/react-query';
+import { redirect } from 'next/navigation';
+import CheckoutForm from './_components/CheckoutForm/CheckoutForm';
+import CheckoutNavigation from './_components/CheckoutNavigation/CheckoutNavigation';
+
+interface PaymentPageProps {
+  searchParams: { [key: string]: string };
+}
+
+export default async function PaymentPage({ searchParams }: PaymentPageProps) {
+  const queryClient = new QueryClient();
+  const { orderId } = searchParams;
+
+  if (!orderId) {
+    redirect('/not-found');
+  }
+
+  await queryClient.prefetchQuery({ queryKey: ['paymentResponse'], queryFn: () => getPayment(orderId) });
+  await queryClient.prefetchQuery({ queryKey: ['addressesData'], queryFn: getAddresses });
```

Comment:
```suggestion
  queryClient.prefetchQuery({ queryKey: ['paymentResponse'], queryFn: () => getPayment(orderId) }),
  queryClient.prefetchQuery({ queryKey: ['addressesData'], queryFn: getAddresses })
```

두 쿼리함수가 의존적이지 않다면 promise.all로 병렬로 처리하는 것은 어떠신가요?